docs/posts/ibis-analytics/index.qmd

```diff
@@ -1,341 +1,1969 @@
-  style Extraction fill:#297932
-  style Transform fill:#ffa500
-  style Upload fill:#4caf50
-  style Analytics fill:#b5e48c
-  style CloudStorage fill:#ffeb3b, stroke:#f44336, stroke-width:2px, stroke-dasharray: 5, 5
-
-  subgraph DataSources[Data sources]
-    GitHub
-    PyPI
-    GoatCounter
-    IbisCI
-  end
-
-  GitHub[GitHub GraphQL API] --> |Ingest with Python as JSON| JSON
-  PyPI[PyPI downloads Google BigQuery dataset] --> |Ingest with Ibis as Parquet| Parquet
-  GoatCounter[GoatCounter REST API or GUI] --> |Ingest as gzipped CSV| CSV
-  IbisCI[Ibis CI Google BigQuery dataset] --> |Ingest with Ibis in DuckDB database file| DuckDB
-
-  subgraph Extraction[Data extraction]
-    JSON
-    Parquet
-    CSV
-    DuckDB
-  end
-
-  Extraction --> |Load sources with Ibis + DuckDB| Transform
-
-  subgraph Transform[Data transformation]
-    PythonScript[Transform with Ibis + DuckDB]
-    GitHubTables[GitHub tables: stars, issues, pulls, forks, watchers, commits]
-    PyPITables[PyPI tables: downloads]
-    CITables[CI tables: jobs, workflows, analysis]
-  end
-
-  PythonScript --> GitHubTables
-  PythonScript --> PyPITables
-  PythonScript --> CITables
-
-  Transform --> |Push to prod| Upload
-
-  subgraph Upload[Load production data]
-    ProductionDB[Production database on MotherDuck]
-    CloudStorage[Backup to Azure Blob Storage]
-  end
-
-  Upload --> |Access data anywhere| Analytics
-
-  subgraph Analytics[Analytics everywhere]
-    PersonalLaptop[Personal laptop]
-    WorkLaptop[Work laptop]
-    Dashboard[Dashboard]
-    Coworkers[Coworkers]
-  end
-```
+## Architecture
+
+The source of truth for the project's architecture [is the repository of
+code](https://github.com/ibis-project/ibis-analytics). A point-in-time
+description and snapshots of the code are below.
+
+### Goals
+
+I had the following goals in mind:
+
+- **Modular:** while this project uses Dagster, it should be easily swappable for other orchestration tools like Airflow or Prefect or Kedro. While this project uses DuckDB, it should be easily swappable for Polars or Clickhouse or any other engine. While this project uses Streamlit, it should be swappable for Quarto dashboards...
+- **Composable:** the project should be easy to understand and extend. It should be easy to add new data sources, new data transformations, new metrics, and new visualizations.
+- **Automated:** the project should be easy to run locally and in production. It should be easy to run the entire pipeline or just a single step. It should be easy to run the pipeline on a schedule or on a pull request.
+- **Iterative:** the project should be easy to iterate on. It should be easy to explore the data, transform it, and visualize it. It should be easy to switch between local and production data sources.
 
-For the most part, you just need a Python environment with `pip` installs including:
+Our open-source tools and freemium services allow us to accomplish these goals
+at little to no cost.
```

Comment:
```suggestion
at little to no cost.
```

Not sure what to say instead here but if we can avoid general statements about how much something costs that'd be best I think, to avoid giving the impression that anything you'll build is going to be low-cost.